src/TemplateBinding.js

```diff
@@ -1058,6 +1106,20 @@
     updateValue: function(value) {
       if (!this.deps.repeat)
         value = [value];
+      else if (Object.prototype.toString.call(value)  === '[object Object]') {
+        if (this.objectArrayProxy !== undefined) {
+          if (value !== this.objectArrayProxy.values) {
```

Comment:
it would be nice to have a test to validate that this check can succeed, in other words, that we aren't allocating a new observer every time.